src/module_3/data_processing.py

```diff
@@ -0,0 +1,46 @@
+import pandas as pd
+import logging
+import os
+
+logging.basicConfig(level=logging.INFO)
+
+DATA_PATH = os.path.join(os.getcwd(), "data", "feature_frame.csv")
+
+
+def load_raw_data() -> pd.DataFrame:
+    logging.info(f"Loading dataset from {DATA_PATH}")
+    try:
+        data = pd.read_csv(DATA_PATH)
+        return data
+    except Exception as e:
+        logging.error(f"An error occurred while loading the dataset: {e}")
+        return None
+
+
+def filter_orders_with_minimum_size(
+    df: pd.DataFrame, min_size: int = 5
+) -> pd.DataFrame:
+    """
+    Filter orders that have at least 5 products
+    """
+    filt = df.query("outcome == 1.0").groupby("order_id").size() >= min_size
+    order_ids = set(filt[filt].index)
+    filtered_df = df[df["order_id"].isin(order_ids)]
```

Comment:
Have you considered doing a join instead isin? Are they the same from a computational perspective? If you want to dig more into this, consider the same case with a big datased in pyspark, and try to read about it.